adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileMSALTests.cs

```diff
@@ -0,0 +1,152 @@
+{
+    /// <summary>
+    /// Contains the core test functionality that will be used by Android and iOS tests
+    /// </summary>
+    public static class CoreMobileMSALTests
+    {
+
+        private const string MSGraph = "https://graph.microsoft.com";
+        private const string UIAutomationAppV2 = "1e245a30-49aa-43eb-b9c1-c11b072cc92b";
+
+        /// <summary>
+        /// Runs through the standard acquire token flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenTest(ITestController controller)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            //Clear Cache
+            controller.Tap("Cache");
+            controller.Tap("clearCache");
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenSilentTest(ITestController controller)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+
+            //Clear Cache
+            controller.Tap("Cache");
+            controller.Tap("clearCache");
+
+            SetInputData(controller, UIAutomationAppV2, "User.Read");
+
+            PerformSignInFlowFlow(controller, user);
+
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+
+            SetInputData(controller, UIAutomationAppV2, "User.Write");
+
+            controller.Tap("acquireTokenSilent");
+
+            //Verify result. Test results are put into a label
+            Assert.IsTrue(controller.GetText("testResult").Contains("Result: Success"));
+        }
+
+        /// <summary>
+        /// Runs through the standard acquire token interactive flow
+        /// </summary>
+        /// <param name="controller">The test framework that will execute the test interaction</param>
+        public static void AcquireTokenADFSvXInteractiveMSALTest(ITestController controller, FederationProvider federationProvider, bool isFederated)
+        {
+            //Get User from Lab
+            var user = controller.GetUser(
```

Comment:
Are all these tests new or have just been copied here for mobile test usage? If not new, can the code be reused from desktop tests?

Reply:
Cant reuse code from desktop tests because the UIs for the desktop apps and mobile apps are all different.